msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -183,36 +181,90 @@ public void onError(final Exception exception) {
 
             ApiDispatcher.getAccounts(command);
 
-        }catch(MsalClientException clientException){
+        } catch (MsalClientException clientException) {
             callback.onError(clientException);
         }
     }
 
+    @Override
+    public ICurrentAccountResult getCurrentAccount() throws InterruptedException, MsalException {
+        throwOnMainThread("getCurrentAccount");
+
+        final ResultFuture<AsyncResult<CurrentAccountResult>> future = new ResultFuture<>();
+
+        getCurrentAccountAsync(new CurrentAccountCallback() {
+            @Override
+            public void onAccountLoaded(@Nullable IAccount activeAccount) {
+                CurrentAccountResult currentAccountResult = new CurrentAccountResult(activeAccount, null, false);
+                future.setResult(new AsyncResult<CurrentAccountResult>(currentAccountResult, null));
+            }
+
+            @Override
+            public void onAccountChanged(@Nullable IAccount priorAccount, @Nullable IAccount currentAccount) {
+                CurrentAccountResult currentAccountResult = new CurrentAccountResult(currentAccount, priorAccount, false);
+                future.setResult(new AsyncResult<CurrentAccountResult>(currentAccountResult, null));
+            }
+
+            @Override
+            public void onError(@NonNull Exception exception) {
+                //TODO: Need to talk to Dome about exception here rather than MsalException
```

Comment:
getCurrentAccountAsync() already returns MsalException. Once you change the param of this interface to MsalException, it should be good.

Reply:
Cool.  Thanks.  I'll update